### ISSUE TITLE
Not compatible with new 16 inch displays

### ISSUE THREAD

**Issue by koholleran** [Kitware/LookingGlassVTKModule#68]
Hi there - I've only recently tried out the looking glass products and I had success using this with a portrait device but now with the newer 16 inch landscape displays there seems to be a new pixel mapping or format that causes the display to no longer work well (fuzzy out of focus with colours spreading out). Is there any interest or plan to include support for larger displays? Error message received is: vtkLookingGlassInterfac:407   WARN| vtkLookingGlassInterface (00000217D298A540): Unrecognized device type: '16_gen3_l', defaulting to setting for 'large' device.

Thanks!

**Comment by psavery** [comment]
@koholleran So, you can look at the various device types we have supported [here](https://github.com/Kitware/LookingGlassVTKModule/blob/6ea6bfb8e7ab27257dbd54358df2815028063921/vtkLookingGlassInterface.cxx#L229-L258).

If we need a new device type, I think we can just add one there. I'm not sure what the ideal quilt size, tiles, and aspect ratio are for new ones.

Note also that fuzzy out of focus can also be caused by the placement of the focal point, and you might need to adjust your camera. See [this note](https://github.com/Kitware/LookingGlassVTKModule/blob/6ea6bfb8e7ab27257dbd54358df2815028063921/vtkLookingGlassInterface.cxx#L479-L487), for example.

**Comment by OndrejSemmler** [comment]
Hi,
I have a problem when displaying:
- Display: Looking Glass 16'' Spatial Display
- Name: LKG-J00188
- Pos: 2560,0
- Dim: 3840,2160

Looking Glass Bridge 2.5.1 1b27c774

```
2025-05-19 12:39:10.767 (   0.171s) [        B58BE740]vtkLookingGlassInterfac:406   WARN| vtkLookingGlassInterface (0x652e09ff7660): Unrecognized device type: '16_gen3_l', defaulting to setting for 'large' device
```

My code:
```
import vtk
from vtk import vtkRenderingLookingGlass

ren = vtk.vtkRenderer()
renWin = vtkRenderingLookingGlass.vtkLookingGlassInterface.CreateLookingGlassRenderWindow(1)
renWin.AddRenderer(ren)

iren = vtk.vtkRenderWindowInteractor()
iren.SetRenderWindow(renWin)

text = vtk.vtkVectorText()
text.SetText("Hello VTK!")
textMapper = vtk.vtkPolyDataMapper()
textMapper.SetInputConnection(text.GetOutputPort())
textActor = vtk.vtkActor()
textActor.SetMapper(textMapper)
ren.AddActor(textActor)

cone = vtk.vtkConeSource()
cone.SetRadius(2)
cone.SetHeight(4)
cone.SetCenter(4,4,2)
cone.SetDirection(0,0,1)
coneMapper = vtk.vtkPolyDataMapper()
coneMapper.SetInputConnection(cone.GetOutputPort())
coneActor = vtk.vtkActor()
coneActor.SetMapper(coneMapper)
ren.AddActor(coneActor)

renWin.Initialize()
ren.ResetCamera()
ren.GetActiveCamera().SetViewAngle(30)

iren.Start()
```

The display shows only a small black window.
On the second display, Looking Glass GO, everything works.

**Comment by OndrejSemmler** [comment]
Also tried latest C++/VTK

![Image](https://github.com/user-attachments/assets/9f86e4b3-0890-4c5b-a5ec-22b9fb1292c8)

**Comment by psavery** [comment]
Yes, we need to update this library to work with the latest 16" displays, as the Looking Glass company updated their API. We are looking for funding to support the labor costs. Let us know if you'd like to help 🙂.

**Comment by OndrejSemmler** [comment]
> Yes, we need to update this library to work with the latest 16" displays, as the Looking Glass company updated their API. We are looking for funding to support the labor costs. Let us know if you'd like to help 🙂.

I have written to the official Kitware contact regarding funding.

As for help, I can contribute but I am not sure how much time I will be able to allocate for this.

**Comment by psavery** [comment]
Thanks for contacting us @OndrejSemmler. We'll get back with you soon 🙂

**Comment by BryanChrisBrown** [comment]
We’re also available for supporting this effort as well, happy to be
involved.
